src/services/urlService.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * @copyright Copyright (c) 2023 Grigorii Shartsev <grigorii.shartsev@nextcloud.com>
+ *
+ * @author Grigorii Shartsev <grigorii.shartsev@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import { generateUrl } from '@nextcloud/router'
+
+/**
+ * Generate a full absolute link with @nextcloud/router.generateUrl
+ *
+ * @see @nextcloud/router.generateUrl
+ * @param {string} url
+ * @param {object} [params] parameters to be replaced into the address
+ * @param {import('@nextcloud/router').UrlOptions} [options] options for the parameter replacement
+ * @param {boolean} options.noRewrite True if you want to force index.php being added
+ * @param {boolean} options.escape Set to false if parameters should not be URL encoded (default true)
+ * @return {string} Full absolute URL
+ */
+export function generateAbsoluteUrl(url, params, options) {
+	// TODO: add this function to @nextcloud/router?
+	const fullPath = generateUrl(url, params, options)
+	if (!IS_DESKTOP) {
```

Comment:
'IS_DESKTOP' is not defined

Reply:
Sorry, I miss-clicked, this PR should be a DRAFT before https://github.com/nextcloud/spreed/pull/9050